### PR TITLE
Improve types for the option function

### PR DESCRIPTION
**Pull request by aabounegm**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +26,8 @@
   "homepage": "https://github.com/V4Fire/Config#readme",
   "devDependencies": {
     "@babel/eslint-parser": "7.16.3",
-    "eslint": "8.3.0"
+    "eslint": "8.3.0",
+    "typescript": "^5.4.2"
```
Comment (kholstinin) [inline package.json]: use typesceript version from core v4 repos https://github.com/V4Fire/Client/blob/v4/package.json#L143

Comment (kholstinin) [reply package.json]: And migrate on yarn

**Review by kholstinin — APPROVED** [review]

File: core/parseArgs.d.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export = parseArgs;
+/**
```
Comment (kobezzza) [inline core/parseArgs.d.ts:2]:
```suggestion

/**
```

File: core/types.d.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export namespace boolean {
+    function coerce(val: any): boolean;
+}
+export namespace number {
+    function coerce(val: any): any;
+    const validate: (number: unknown) => boolean;
+}
+export namespace json {
```
Comment (kobezzza) [inline core/types.d.ts:8]:
```suggestion

export namespace json {
```

File: core/types.d.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export namespace boolean {
+    function coerce(val: any): boolean;
+}
```
Comment (kobezzza) [inline core/types.d.ts:3]:
```suggestion
}

```

File: options/computed.d.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @param {Function} handler
+ * @returns {boolean}
+ */
+export function isComputed(handler: Function): boolean;
+/**
```
Comment (kobezzza) [inline options/computed.d.ts:6]:
```suggestion

/**
```

File: options/option.d.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+export = option;
+/**
```
Comment (kobezzza) [inline options/option.d.ts:2]:
```suggestion

/**
```

**Review by kobezzza — APPROVED** [review]

**Review by shining-mind — APPROVED** [review]

**Pull request merged**